src/test/java/seedu/address/logic/commands/ListAttendanceCommandTest.java

```diff
@@ -0,0 +1,515 @@
+
+        // same values -> returns true
+        ListAttendanceCommand alexT01CommandCopy =
+                new ListAttendanceCommand(null, ALEX.getStudentId(), studentRelatedWeek);
+        assertTrue(alexCommand.equals(alexT01CommandCopy));
+        ListAttendanceCommand t01CommandCopy = new ListAttendanceCommand(T01.getTutorialName(), null, week);
+        assertTrue(t01Command.equals(t01CommandCopy));
+
+        // different types -> returns false
+        assertFalse(alexCommand.equals(1));
+        assertFalse(t01Command.equals(1));
+
+        // null -> returns false
+        assertFalse(alexCommand.equals(null));
+        assertFalse(t01Command.equals(null));
+
+        // different values -> returns false
+        assertFalse(alexCommand.equals(bernardCommand));
+        assertFalse(alexCommand.equals(t01Command));
+        assertFalse(t01Command.equals(t02Command));
+    }
+
+    @Test
+    public void execute_validStudentInput_success() throws CommandException {
+        String expectedMessage = String.format(ListAttendanceCommand.MESSAGE_SUCCESS_STUDENT, ALEX.getName());
+        ListAttendanceCommand command =
+                new ListAttendanceCommand(null, ALEX.getStudentId(), studentRelatedWeek);
+        ModelStubAcceptingStudentId model = new ModelStubAcceptingStudentId();
+        CommandResult result = command.execute(model);
+        assertEquals(expectedMessage, result.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_validTutorialNameWeekInput_success() throws CommandException {
+        String expectedMessage = String.format(ListAttendanceCommand.MESSAGE_SUCCESS, T01.getTutorialName(), validWeek);
+        ListAttendanceCommand command =
+                new ListAttendanceCommand(T01.getTutorialName(), null, validWeek);
+        ModelStubAcceptingTutorialName model = new ModelStubAcceptingTutorialName();
+        CommandResult result = command.execute(model);
+        assertEquals(expectedMessage, result.getFeedbackToUser());
+        assertEquals(validWeek, result.getAttendanceWeek());
+    }
+
+    @Test
+    public void execute_invalidStudentInput_failure() {
+        String expectedMessage = GradeCommand.MESSAGE_STUDENT_NOT_FOUND;
+        ListAttendanceCommand command =
+                new ListAttendanceCommand(null, BERNARD.getStudentId(), studentRelatedWeek);
+        ModelStubAcceptingStudentId model = new ModelStubAcceptingStudentId();
+        try {
+            command.execute(model);
+        } catch (CommandException ce) {
+            assertEquals(expectedMessage, ce.getMessage());
+        }
+    }
+
+    @Test
+    public void execute_invalidTutorialNameInput_failure() {
+        String expectedMessage = Messages.MESSAGE_TUTORIAL_NOT_FOUND;
+        ListAttendanceCommand command =
+                new ListAttendanceCommand(T02.getTutorialName(), null, validWeek);
+        ModelStubAcceptingTutorialName model = new ModelStubAcceptingTutorialName();
+        try {
```

Comment:
same as below, should we use assertThrows instead? i think it also helps to check the type of error thrown

Reply:
Thanks! Will make the changes